### DISCUSSION TITLE
RateLimiter unexpected behaviour with ClockMock

### DISCUSSION THREAD

**Discussion by grga-giscloud** [symfony/symfony#51873] [Q&A] [answered]
Hi, I am testing following code:

```
<?php
declare(strict_types=1);
namespace App\Tests\Unit\Service;
use PHPUnit\Framework\TestCase;
use Symfony\Bridge\PhpUnit\ClockMock;
use Symfony\Component\RateLimiter\RateLimiterFactory;
use Symfony\Component\RateLimiter\Storage\InMemoryStorage;
/**
 * @group time-sensitive
 */
final class SimpleTest extends TestCase
{
    public function testFoo(): void
    {
        ClockMock::register(InMemoryStorage::class);
            $rateLimiterFactory = new RateLimiterFactory([
            'id' => 'id',
            'policy' => 'fixed_window',
            'limit' => 2,
            'interval' => '5 seconds',
        ], new InMemoryStorage());
            $limiter = $rateLimiterFactory->create('foo');
            for ($i = 0; $i < 3; ++$i) {
            static::assertTrue($limiter->consume(1)->isAccepted());
            sleep(4);
        }
    }
}
```
The test runs instantly and fails, which I find unexpected.

If I remove comment `@group time-sensitive`, test will actually run for some time (~12sec) and will pass.

I thought I made some mistake with mocking time, but, if I leave comment `@group time-sensitive`, but change the last `sleep(4)` to `sleep(5)` the test runs instantly and passes, which is expected and makes me think that mocking time isn't done completely incorrectly.

Could you help me understand why initial test fails and help me fix it?

**Comment by stof** [comment, marked as answer]
You should be registering a ClockMock for `FixedWindowLimiter::class` to mock function calls done in the namespace of the FixedWindowLimiter

**Reply by grga-giscloud** [reply]
Thanks, that solves it!

Though I am not sure why then when I ran original code with `sleep(5)` I got the test to pass?
